File: FluidStudio/FluidStudio/Physics/CSGBoundary/CSGBoundaryScene.cs
using PG.Core.Math;
using PG.Scene;
using Reactive.Bindings;
using CreateLabels = PG.CSGBoundarySceneCreateLabels;
using UpdateLabels = PG.CSGBoundarySceneUpdateLabels;

namespace FluidStudio.Physics
{
    public class CSGBoundaryScene : IPhysicsScene
    {
        public int Id { get; private set; }

        public string Name { get; private set; }

        public Box3d BoundingBox { get; private set; }

        public IPhysicsScene Parent { get; private set; }

        private readonly SceneList scenes;

        public CSGBoundaryScene(SolverScene parent, SceneList world, string name, Box3d bb)
        {
            this.Parent = parent;
            this.scenes = world;
            Create(name, bb);
        }
        
        public void Create(string name, Box3d bb)
        {
            var command = new PG.CLI.PhysicsCommand(CreateLabels.CommandNameLabel);
            command.Execute(scenes.Adapter);
            this.Id = command.GetResult<int>(CreateLabels.NewIdLabel);
            this.Name = name;
            this.BoundingBox = bb;
            Update(name, bb);
        }

        public void Update(string name, Box3d bb)
        {
            this.BoundingBox = bb;
            var command = new PG.CLI.PhysicsCommand(UpdateLabels.CommandNameLabel);
            command.SetArg(UpdateLabels.IdLabel, Id);
            command.SetArg(UpdateLabels.NameLabel, name);
            command.SetArg(UpdateLabels.BoundingBoxLabel, bb);
            command.Execute(scenes.Adapter);
            this.Name = name;
        }

        public void Delete()
        {
            scenes.Delete(this.Id);
        }
    }
}

File: Blender/CrystalPython/physics/csg_boundary_scene.py
import physics_labels
from scene import Scene
from CrystalPLI import *

class CSGBoundaryScene :
    def __init__(self, scene) :
        self.scene = scene
        self.id = 0
        self.name = ""
        self.bounding_box = Box3dd()

    def create(self) :
        create_physics_command(physics_labels.CSGBoundarySceneCreateCommand.CommandNameLabel)
        execute_command(self.scene.world)
        self.id = get_result_int(physics_labels.FluidSceneCreateCommand.NewIdLabel)

    def send(self) :
        create_physics_command(physics_labels.CSGBoundarySceneUpdateCommand.CommandNameLabel)
        set_arg_int(physics_labels.CSGBoundarySceneUpdateCommand.IdLabel, self.id)
        set_arg_string(physics_labels.CSGBoundarySceneUpdateCommand.NameLabel, self.name)
        set_arg_box3dd(physics_labels.CSGBoundarySceneUpdateCommand.BoundingBoxLabel, self.bounding_box)
        is_ok = execute_command(self.scene.world)
        return is_ok
    
    def delete(self) :
        self.scene.delete(self.id, False)
